www_src/components/text-input/text-input.jsx

```diff
@@ -28,12 +30,15 @@ var TextInput = React.createClass({
     }
   },
   render: function () {
+    var linkState = this.props.linkState || this.linkState;
+    var valueLink = this.valueLink = linkState(this.props.id);
```

Comment:
Does a local var need to be created for some reason? Seems like just using this.valueLink should be fine...